### ISSUE TITLE
[BUG] Error: UNKNOWN: unknown error, open '\\wsl$\rancher-desktop-data\etc\hosts'

### ISSUE THREAD

**Issue by hencoappel** [rancher-sandbox/rancher-desktop#7786]
### Actual Behavior

After a reboot while having some containers running, Rancher Desktop fails to start with the following issue:
> Kubernetes Error
> Error: UNKNOWN: unknown error, open '\\wsl$\rancher-desktop-data\etc\hosts'

Confusingly I've got Kubernetes disabled.

### Steps to Reproduce

Not really sure. This seems to be a regular occurrence in our team.
We run several containers with mounted volumes using docker compose.
It seems to be after a system reboot we run into this issue and the only way to resolve it seems 

### Result

![image](https://github.com/user-attachments/assets/f974d4d9-b037-4d74-877a-74679703b2a2)

### Expected Behavior

Rancher to start up successfully

### Additional Information

[logs.zip](https://github.com/user-attachments/files/17812057/logs.zip)
I've attached all log files after enabling debug.

The only way to resolve the issues seems to be to reset Kubernetes and factory reset Rancher from the troubleshooting menu.
This is very frustrating if you have to do it multiple times a week.

### Rancher Desktop Version

1.16.0

### Rancher Desktop K8s Version

NA

### Which container engine are you using?

moby (docker cli)

### What operating system are you using?

Windows

### Operating System / Build Version

Windows 11 23H2 22631.4460

### What CPU architecture are you using?

x64

### Linux only: what package format did you use to install Rancher Desktop?

None

**Comment by hencoappel** [comment]
Similar to this previously closed issue: https://github.com/rancher-sandbox/rancher-desktop/issues/6011

**Comment by hencoappel** [comment]
Anyone have a chance to look into this?

**Comment by creesch** [comment]
> Anyone have a chance to look into this?

Not really an expert by any means, but I got the same error and since it is reporting an issue with the wsl environment so I effectively did a hard reset and that "fixed" it. In quotes because it also threw away my containers. Which I was fine with as I could easily recreate them anyway, but obviously that might not be a solution for you. 

```
wsl --shutdown
wsl --unregister rancher-desktop
wsl --unregister rancher-desktop-data
```

**Comment by hencoappel** [comment]
Hi @creesch, yes I've discovered that you can do this to get back to a stable state, but the issue is that it happens at all. This can happen fairly frequently and having to wipe all containers and images every time is quite frustrating.

**Comment by creesch** [comment]
> Hi @creesch, yes I've discovered that you can do this to get back to a stable state, but the issue is that it happens at all. This can happen fairly frequently and having to wipe all containers and images every time is quite frustrating.

I agree, I am just posting it here is a workaround for people who google it (like I did) until a fix is implemented.

**Comment by hencoappel** [comment]
Any chance someone has had a look at this? It's still plaguing some of my colleagues.

**Comment by jason-vignochi** [comment]
Is this still up in the air? I have had this issue over and over and can't find a workaround.

**Comment by hencoappel** [comment]
We've not really experienced it recently that I'm aware of.

**Comment by slaknn** [comment]
Not sure if it's related, but I got this error after trying to compress the data distro with diskpart.  I forgot to detach it, so it was still mounted read only.  After detaching the vhdx, Rancher Desktop loaded with no issues.

**Comment by whereisaaron** [comment]
Windows 11 (25H2 latest) crashed and now Rancher Desktop (1.23.1) was unrecoverable (even after full uninstall, reboot, and reinstall) with this same error other resport: `Error: ECONNRESET: connection reset by peer, open '\\wsl$\rancher-desktop-data\etc\hosts'`.

TL;DR Rancher Desktop doesn't realise WSL VHDs are available to non-Rancher processes. Had to make sure not other non-Rancher process was touching the rancher VHDs, in order for Rancher to reinstall and set-up correctly.

Previous attempts:
-------------------
'Diagnostics' screen says no problems.

Tried "Reset Kubernetes" and "Factory Reset" in app --> same error.

Tried full exiting Rancher Desktop, and restarting Rancher Desktop, get install screen, choose NO Kubernetes, and dockerd -> same error.
```
wsl --shutdown
wsl --unregister rancher-desktop
wsl --unregister rancher-desktop-data
```

Checked logs and they are all empty, except `wsl.log`:
```
2026-07-27T12:03:39.777Z: WSL failed to execute wsl.exe --import rancher-desktop-data C:\Users\username\AppData\Local\rancher-desktop\distro-data C:\Users\username\AppData\Local\Temp\rd-distro-22UJoC\distro.tar --version 2: Error: wsl.exe exited with code 4294967295

2026-07-27T12:03:39.777Z: Error registering data distribution: Error: wsl.exe exited with code 4294967295

2026-07-27T12:03:39.824Z: WSL failed to execute wsl.exe --unregister rancher-desktop-data: Error: wsl.exe exited with code 4294967295
```

Did full uninstall, reboot, then:
```
wsl --shutdown
wsl --unregister rancher-desktop
wsl --unregister rancher-desktop-data
```
Then download Rancher Desktop 1.23.1 and installed fresh --> same error

As best I can tell the app/installer cannot recreate the `rancher-desktop-data` distro even after uninstall. I checked the location and there is a `ext4.vhdx` file there with.
```
2026-07-27T12:33:30.387Z: Running: wsl.exe --import rancher-desktop-data C:\Users\username\AppData\Local\rancher-desktop\distro-data C:\Users\username\AppData\Local\Temp\rd-distro-RyJgIt\distro.tar --version 2
The supplied install location is already in use.
Error code: Wsl/Service/RegisterDistro/ERROR_FILE_EXISTS
2026-07-27T12:33:30.535Z: Running: wsl.exe --unregister rancher-desktop-data
There is no distribution with the supplied name.
Error code: Wsl/Service/WSL_E_DISTRO_NOT_FOUND
```

I fully uninstalled again and found that Rancher Desktop uninstall doesn't clean up of the data files! Do I deleted them myself but go an in use error for the `distro-data` `ext4.vhdx` I traced this to a back-up application currently backing up WSL2 files. By stopping the back-up I was able to get Rancher Desktop re-installed.

Then I got a different error: `Error: EINVAL: invalid argument, open '\\wsl.localhost\rancher-desktop\etc\conf.d\moproxy'`.

So then I did this again:
```
wsl --shutdown
wsl --unregister rancher-desktop
wsl --unregister rancher-desktop-data
```
And now appears working again.
